libraries/chain/include/graphene/chain/protocol/transaction.hpp

```diff
@@ -176,19 +181,36 @@ namespace graphene { namespace chain {
        *       otherwise, the @ref chain_id parameter will be ignored, and
        *       @ref signees will be returned directly.
        */
-      const flat_set<public_key_type>& get_signature_keys( const chain_id_type& chain_id )const;
+      virtual const flat_set<public_key_type>& get_signature_keys( const chain_id_type& chain_id )const;
 
       /** Signatures */
       vector<signature_type> signatures;
 
-      /** Public keys extracted from signatures */
-      mutable flat_set<public_key_type> signees;
+      /** Removes all operations and signatures */
+      void clear() { operations.clear(); signatures.clear(); }
```

Comment:
signed_transaction recomputes signees on every call, so no need to clear.
precomputable_transaction uses the cached _signees but doesn't allow modification, so no need to clear either.

Reply:
Got it. Thanks!

>signed_transaction recomputes signees on every call

I would expect some modifications on code to hedge the performance loss due to this (I haven't thoroughly review all the changes, perhaps you've done it already, I'll re-check later).

Reply:
The goal would be to avoid any performance loss due to this.
During normal operation, signees are only computed once, so it shouldn't matter. Still working on pending transactions and fork database.